core/io/resource.h

```diff
@@ -83,11 +83,12 @@ class Resource : public RefCounted {
 public:
 	static Node *(*_get_local_scene_func)(); //used by editor
 	static void (*_update_configuration_warning)(); //used by editor
+	static Ref<Resource> get_remap_resource(const Ref<Resource> &p_resource, HashMap<Ref<Resource>, Ref<Resource>> &remap_cache, const Ref<Resource> &p_fallback, Node *p_for_scene);
 
 	void update_configuration_warning();
 	virtual bool editor_can_reload_from_file();
 	virtual void reset_state(); //for resources that use variable amount of properties, either via _validate_property or _get_property_list, this function needs to be implemented to correctly clear state
-	virtual Error copy_from(const Ref<Resource> &p_resource);
+	virtual Error copy_from(const Ref<Resource> &p_resource, HashMap<Ref<Resource>, Ref<Resource>> &remap_cache, bool p_keep_local_to_scene_resource_instance = false);
```

Comment:
If this logic is only used in PackedScene, it should be in PackedScene IMO.

Reply:
It is for `Resource` with `local_to_scene` enabled, used in `Resource::get_remap_resource()`.